messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -639,7 +639,9 @@ public <R> void resetTokens(@Nonnull TrackingToken startPosition, R resetContext
             eventHandlerInvoker().performReset(resetContext);
 
             for (int i = 0; i < tokens.length; i++) {
-                tokenStore.storeToken(ReplayToken.createReplayToken(tokens[i], startPosition), getName(), segments[i]);
+                tokenStore.storeToken(ReplayToken.createReplayToken(tokens[i], startPosition, resetContext),
```

Comment:
Nice to see how easy this addition is from the perspective of the Event Processor. :-)